wp-admin/js/widgets/media-widgets.js

```diff
@@ -603,11 +596,7 @@ wp.mediaWidgets = ( function( $ ) {
 				var attachment = {}, state = mediaFrame.state();
 
 				// Update cached attachment object to avoid having to re-fetch. This also triggers re-rendering of preview.
-				if ( 'embed' === state.get( 'id' ) ) {
-					_.extend( attachment, { id: 0 }, state.props.toJSON() );
-				} else {
-					_.extend( attachment, state.get( 'selection' ).first().toJSON() );
-				}
+				_.extend( attachment, state.get( 'selection' ).first().toJSON() );
```

Comment:
The `embed` state is needed for when the user inserts an external URL, as opposed to selecting from the media library.